Remove code guarded by USE_CTFONTHASTABLE

vitorroriz opened the pull request:
#### e655079064af83588a5c9021d7ff449ba86da050
<pre>
Remove code guarded by USE_CTFONTHASTABLE
<a href="https://bugs.webkit.org/show_bug.cgi?id=276104">https://bugs.webkit.org/show_bug.cgi?id=276104</a>
<a href="https://rdar.apple.com/130928732">rdar://130928732</a>

Reviewed by Brent Fulgham.

We no longer need this since USE(CTFONTHASTABLE)
is always true for our builds now.

We are also removing the wrapper `fontHasTable` as
it is now exactly what the lower level CT function
`fontHasTable` does. Wrapping seems unnecessary as
the function is just used locally and if any unlikely
change would occur with the CT API it would be fairly
simple to update it.

* Source/WTF/wtf/PlatformUse.h:
* Source/WebCore/platform/graphics/Font.h:
* Source/WebCore/platform/graphics/coretext/FontCoreText.cpp:
(WebCore::fontHasEitherTable):
(WebCore::Font::platformInit):
(WebCore::fontHasTable): Deleted.

Canonical link: <a href="https://commits.webkit.org/280564@main">https://commits.webkit.org/280564@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/eba9e45e731104967538edc48442e12b15688d91

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/56982 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/36310 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/55/builds/9457 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/60602 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/59/builds/7425 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/59110 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/43933 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/61/builds/7615 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/46146 "Passed tests") | [✅ 🧪 wincairo-tests](https://ews-build.webkit.org/#/builders/60/builds/5215 "Passed tests") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/59012 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/47/builds/34096 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/49180 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/27006 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/42/builds/30876 "Passed tests") | [❌ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/64/builds/6511 "Found 1 new test failure: imported/w3c/web-platform-tests/html/semantics/interactive-elements/the-dialog-element/dialog-autofocus.html (failure)") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/6430 "Built successfully") | 
| [✅ 🛠 🧪 jsc](https://ews-build.webkit.org/#/builders/20/builds/50074 "Built successfully and passed tests") | [  ~~🧪 api-ios~~](https://ews-build.webkit.org/#/builders/13/builds/52833 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/63/builds/6782 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/62283 "Built successfully") | 
| [✅ 🛠 🧪 jsc-arm64](https://ews-build.webkit.org/#/builders/12/builds/56223 "Built successfully and passed tests") | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/87/builds/895 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/62/builds/6886 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/53404 "Passed tests") | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/86/builds/898 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/49235 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/53441 "Passed tests") | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/12597 "Built successfully and passed tests") | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/88/builds/757 "Passed tests") | | [✅ 🛠 jsc-armv7](https://ews-build.webkit.org/#/builders/35/builds/77984 "Built successfully") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/44/builds/32139 "Built successfully") | | [✅ 🧪 jsc-armv7-tests](https://ews-build.webkit.org/#/builders/25/builds/12926 "Passed tests") | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/46/builds/33224 "Built successfully") | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/43/builds/34309 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/45/builds/32970 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

brentfulgham commented on Source/WebCore/platform/graphics/coretext/FontCoreText.cpp:
```diff
@@ -109,38 +109,12 @@ static bool isAhemFont(CFStringRef familyName)
 
 bool fontHasTable(CTFontRef ctFont, unsigned tableTag)
```
Do we need this WebKit wrapper around this CoreText call? It is only used in CoreText code, so why wrap it in a C++ method? Probably gets inlined away, but seems like unnecessary indirection.

vitorroriz replied:
Yes. I was thinking to remove that but I wasn't sure if I should remove a wrapper. Anyway, if unlikely something changes with the CT function it wouldn't be much work to change it accordingly. I can do it.

brentfulgham commented on Source/WebCore/platform/graphics/coretext/FontCoreText.cpp:
```diff
@@ -109,38 +109,12 @@ static bool isAhemFont(CFStringRef familyName)
 bool fontHasEitherTable(CTFontRef ctFont, unsigned tableTag1, unsigned tableTag2)
 {
-#if USE(CTFONTHASTABLE)
     return fontHasTable(ctFont, tableTag1) || fontHasTable(ctFont, tableTag2);
```
Then these could just both be `CTFontHasTable(...)`.

brentfulgham reviewed (approved):
I think this is great, but propose ditching our custom C++ method in favor of just calling the CoreText function.

brentfulgham reviewed (approved):
Beautiful! So much cleaner!

vitorroriz committed:
Remove code guarded by USE_CTFONTHASTABLE
https://bugs.webkit.org/show_bug.cgi?id=276104
rdar://130928732

Reviewed by Brent Fulgham.

We no longer need this since USE(CTFONTHASTABLE)
is always true for our builds now.

We are also removing the wrapper `fontHasTable` as
it is now exactly what the lower level CT function
`fontHasTable` does. Wrapping seems unnecessary as
the function is just used locally and if any unlikely
change would occur with the CT API it would be fairly
simple to update it.

* Source/WTF/wtf/PlatformUse.h:
* Source/WebCore/platform/graphics/Font.h:
* Source/WebCore/platform/graphics/coretext/FontCoreText.cpp:
(WebCore::fontHasEitherTable):
(WebCore::Font::platformInit):
(WebCore::fontHasTable): Deleted.

Canonical link: https://commits.webkit.org/280564@main
